app/models/generic_event/journey_cancel.rb

```diff
@@ -0,0 +1,9 @@
+class GenericEvent
+  class JourneyCancel < GenericEvent
+    include JourneyEventValidations
+
+    def trigger
+      eventable.cancel
+    end
+  end
+end
```

Comment:
```
class GenericEvent
  class subclass < GenericEvent
```
this pattern drives me crazy :D
My brain goes into loop,
do you have a article that explain it?

Reply:
what's happen if we just create something like
`JourneyCancel < GenericEvent`
but not as a subclass of GenericEvent?
We cound add a module too, to add a scope